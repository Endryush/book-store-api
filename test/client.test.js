import request from 'supertest';
import app from '../src/server'

const validPayload = {
  name: "clientNAme",
  email: "email@email.com",
  phone: "454545",
  password: "senhaforte",
  address: "rua xcv, 0 - lalala"
}

describe('Testing API on /client', () => {
  test('POST /client invalid Payload', async () => {
    const response = await request(app)
      .post('/api/client')
      .send({})

      expect(response.status).toBe(400)
  })

  test ('POST /client with validPayload', async () => {
    const response = await request(app)
      .post('/api/client')
      .send(validPayload)

    expect(response.status).toBe(201)
  })
})