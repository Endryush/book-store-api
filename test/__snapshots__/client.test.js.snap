// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing API GET on /client Valid client 1`] = `
{
  "address": "rua xcv, 0 - lalala",
  "email": "email@email.com",
  "id": 0,
  "name": "clientNAme",
  "password": "senhaforte",
  "phone": "454545",
}
`;
